### DISCUSSION TITLE
rbenv install 3.2.2 fails with cannot load such file -- rubygems/gem_runner (LoadError)

### DISCUSSION THREAD

**Discussion by bigos** [rbenv/ruby-build#2205] [Build failures] [answered]
### Steps to reproduce the behavior

I expect to install new Ruby

### Expected vs. actual behavior

Installation fails

### Logs

> 
> compiling parser.c
> compiling ruby_objs.c
> linking shared-object json/ext/parser.so
> compiling unescape.c
> make[2]: Leaving directory '/tmp/ruby-build.20230612155657.206151.QNtlZi/ruby-3.2.2/ext/json/parser'
> linking shared-object rbs_extension.so
> linking shared-object ripper.so
> make[2]: Leaving directory '/tmp/ruby-build.20230612155657.206151.QNtlZi/ruby-3.2.2/ext/ripper'
> make[2]: Leaving directory '/tmp/ruby-build.20230612155657.206151.QNtlZi/ruby-3.2.2/.bundle/gems/rbs-2.8.2/ext/rbs_extension'
> linking shared-object json/ext/generator.so
> make[2]: Leaving directory '/tmp/ruby-build.20230612155657.206151.QNtlZi/ruby-3.2.2/ext/json/generator'
> make[2]: Entering directory '/tmp/ruby-build.20230612155657.206151.QNtlZi/ruby-3.2.2'
> `RubyGems' were not loaded.
> `error_highlight' was not loaded.
> `did_you_mean' was not loaded.
> `syntax_suggest' was not loaded.
> /home/jacekp/.rbenv/versions/3.2.1/bin/gem:8:in `require': cannot load such file -- rubygems/gem_runner (LoadError)
> 	from /home/jacekp/.rbenv/versions/3.2.1/bin/gem:8:in `<main>'
> make[2]: *** [uncommon.mk:1224: .revision.time] Error 1
> make[2]: Leaving directory '/tmp/ruby-build.20230612155657.206151.QNtlZi/ruby-3.2.2'
> make[1]: *** [exts.mk:93: ruby] Error 2
> make[1]: Leaving directory '/tmp/ruby-build.20230612155657.206151.QNtlZi/ruby-3.2.2'
> make: *** [uncommon.mk:331: build-ext] Error 2
> 
>

**Comment by bigos** [comment]
I have spotted something odd, I use ruby version
```
$ ruby --version
ruby 2.4.5p335 (2018-10-18 revision 65137) [x86_64-linux]
```
but the error points to another

/home/jacekp/.rbenv/versions/3.2.1/bin/gem:8:in `require': cannot load such file -- rubygems/gem_runner (LoadError)

**Comment by bigos** [comment]
rbenv uninstall 3.2.1
and
rbenv install 3.2.1
works

**Comment by bigos** [comment]
but trying to install ruby 3.2.2
fails again
`/home/jacekp/.rbenv/versions/3.2.1/bin/gem:8:in `require': cannot load such file -- rubygems/gem_runner (LoadError)`

**Comment by bigos** [comment, marked as answer]
rbenv global 
was returning 3.21

I will change to system 
`rbenv global system`

and try again

**Reply by mislav** [reply]
Thanks for sharing your solution

**Comment by bigos** [comment]
it worked
